spec/function.dd

```diff
@@ -2749,17 +2749,56 @@ $(H3 $(LNAME2 trusted-functions, Trusted Functions))
 
         $(P Trusted functions are marked with the $(CODE @trusted) attribute.)
 
-        $(P Trusted functions are guaranteed by the programmer to not exhibit
-        any undefined behavior if called by a safe function.
-        Generally, trusted functions should be kept small so that they are
-        easier to manually verify.
+        $(P Trusted functions are guaranteed to not exhibit any undefined
+        behavior if called by a safe function. Furthermore, calls to trusted
+        functions cannot lead to undefined behavior in @safe code that is
+        executed afterwards. It is the responsibility of the programmer to
+        ensure that these guarantees are upheld.
         )
 
+        $(P Example:)
+
+        ---
+        immutable(int)* f(int* p) @trusted
+        {
+            version (none) p[2] = 13;
+            // Invalid. p[2] is out of bounds. This line would exhibit undefined
+            // behavior.
+
+            version (none) p[1] = 13;
+            // Invalid. In this program, p[1] happens to be in-bounds, so the
+            // line would not exhibit undefined behavior, but a trusted function
+            // is not allowed to rely on this.
+
+            version (none) return cast(immutable) p;
+            // Invalid. @safe code still has mutable access and could trigger
+            // undefined behavior by overwriting the value later on.
+
+            int* p2 = new int;
+            *p2 = 42;
+            return cast(immutable) p2;
+            // Valid. After f returns, no mutable aliases of p2 can exist.
+        }
+
+        void main() @safe
```

Comment:
I think `main` should go at the top, followed by `f`. This is because the comments explaining what works and what doesn't in `f` require the reader to have read the call site first.

Reply:
> I think `main` should go at the top

I put `f` at the top because its validity is independent of whatever `main` does. But either way is fine with me. If you want `main` at the top, just say the word and I'll move it.